### PR TITLE
feat: add separate celery queue for usage reports

### PR DESCRIPTION
**Pull request by raquelmsmith**
## Problem

Usage reports cause a big spike in celery tasks, which then trigger an alert for the redis queue depth. We can move these to a separate queue with no alert so that they don't trigger it.

## Changes

- [x] Create a `usage_report` queue
- [x] Add metrics for this queue so we can see when they are running (if there are issues with them running the first thing I look at is celery)

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: posthog/settings/sentry.py [hunk 1]
```diff
@@ -109,7 +108,10 @@ def traces_sampler(sampling_context: dict) -> float:
         task = sampling_context.get("celery_job", {}).get("task")
         if task == "posthog.celery.redis_heartbeat":
             return 0.0001  # 0.01%
-        if task == "posthog.celery.redis_celery_queue_depth":
+        if (
+            task == "posthog.celery.redis_celery_queue_depth"
+            or task == "posthog.celery.redis_celery_queue_depth_usage_reports"
+        ):
```
Comment (webjunkie) [inline posthog/settings/sentry.py:114]: Can also now write `if task in ("...", "...")` for better readability.

**Review by webjunkie — APPROVED** [review]
Looks good!

File: posthog/settings/sentry.py (hunk 1 above)
Comment (benjackwhite) [reply posthog/settings/sentry.py]: Updated for this

File: posthog/tasks/usage_report.py [hunk 2]
```diff
@@ -63,6 +64,14 @@
 QUERY_RETRY_DELAY = 1
 QUERY_RETRY_BACKOFF = 2
 
+USAGE_REPORT_TASK_KWARGS = dict(
+    queue=CeleryQueue.USAGE_REPORTS.value,
```
Comment (benjackwhite) [inline posthog/tasks/usage_report.py:68]: @raquelmsmith just FYI - modified this to have each task use the queue. It doesn't work to only have the parent task on its own. The task that get created also need to specify their intended queue

**Review by benjackwhite — APPROVED** [review]

File: posthog/tasks/usage_report.py (hunk 2 above)
Comment (raquelmsmith) [reply posthog/tasks/usage_report.py:68]: Oh that makes sense. Thanks!

**Pull request merged**